pennylane/measurements/sample.py

```diff
@@ -182,9 +182,25 @@ def __init__(self, obs=None, wires=None, eigvals=None, id=None):
 
         super().__init__(obs=obs, wires=wires, eigvals=eigvals, id=id)
 
-    @property
-    def return_type(self):
-        return Sample
+    return_type = Sample
+
+    @classmethod
+    def _abstract_eval(
+        cls, n_wires: Optional[int] = None, shots: Optional[int] = None, num_device_wires: int = 0
+    ):
+        if shots is None:
+            raise ValueError("finite shots are required to SampleMP")
+        dtype = float if n_wires is None else int
```

Comment:
Is this correct? And what happens when we sample from all device wires? In that case `n_wires is None` but we are sampling integers, no? :thinking: 
```suggestion
        dtype = float if n_wires is None else int # If no wires are given, the output is sampled from eigvals
```

Reply:
So `n_wires is None` corresponds to "we are measuring an observable".  See #5369 and #5370 . We will be standardizing sampling an observable to be returning floats.

If `n_wires=0`, then we are broadcasting over all device wires.

I should probably get around to adding some more documentation.